src/conn/conn_layered.c

```diff
@@ -232,6 +233,37 @@ __disagg_pick_up_checkpoint(WT_SESSION_IMPL *session, uint64_t checkpoint_id)
     return (ret);
 }
 
+/*
+ * __disagg_pick_up_checkpoint_meta --
+ *     Pick up a new checkpoint from a metadata string.
+ */
+static int
+__disagg_pick_up_checkpoint_meta(WT_SESSION_IMPL *session, const char *meta, size_t len)
+{
+    WT_CONFIG_ITEM cval;
+    WT_DECL_RET;
+    uint64_t checkpoint_id, metadata_lsn;
+    char *meta_buf;
+
+    /* Need to ensure that the meta argument is NUL-terminated. */
+    WT_ERR(__wt_calloc_def(session, len + 1, &meta_buf));
+    memcpy(meta_buf, meta, len);
+    meta_buf[len] = '\0';
```

Comment:
Even if you're going to ignore my other comment, you can get rid of the explicit NUL termination, as the calloc guarantees that.